Fix avi dump feature by using render buffer with option to use output buffer.

LunaMoo opened the pull request:
Fixes #10632 as tested on 20 fps game.

Fixed with render buffer now:
 - uses render resolution which can get very big very quickly,
 - at the same time will be much faster when forcing x1 res like software rendering.

Changing to output buffer means we're now recording also:
- shader effects(which is good I guess),
- on screen text(possibly annoying, but osm for savestates disabled make it very usable),
- touch screen controls(some might want it, some might not, the feature is not enabled on mobiles anyway so not a big deal).


 ~~One other downside I found is that software rendering isn't recorded correctly with output buffer at least if the game draws by manually copying to vram like that Darkstalkers game which can't render in hardware renderer correctly. It recorded ok with display framebuffer, so maybe giving a choice of buffer for recording would be a nice option, through that's really an edge case.~~ - software renderer recording overall works better with render buffer, through while testing now, it recorded fine even with output buffer, not sure why.

LunaMoo committed:
Fix avi dump feature by using output buffer.

hrydgard commented:
Hm, not sure if that's a good thing, although I guess if you want a clean image, you'll just turn all the overlay stuff off...

LunaMoo committed:
Block savestates OSM when video frames are dumped.
Also make the dump messages shorter,
especially dump start since it will be recorded.

LunaMoo commented:
Since avi dump cooperates nicely with savestates I blocked OSM for those when video is being dumped also made the OSM for dump shorter(especially start) to reduce annoyance as it will record into video.

LunaMoo committed:
Fix avi dump via render buffer instead,
with an option to use output buffer since recording with
post process effects will often be desirable.

hrydgard commented on UI/GameSettingsScreen.cpp:
```diff
@@ -813,6 +813,7 @@ void GameSettingsScreen::CreateViews() {
 #if defined(_WIN32) || (defined(USING_QT_UI) && !defined(MOBILE_DEVICE))
 	systemSettings->Add(new CheckBox(&g_Config.bDumpFrames, sy->T("Record Display")));
 	systemSettings->Add(new CheckBox(&g_Config.bUseFFV1, sy->T("Use Lossless Video Codec (FFV1)")));
+	systemSettings->Add(new CheckBox(&g_Config.bDumpVideoOutput, sy->T("Use output buffer(with overlay) for recording")));
```
add a space before (with

LunaMoo committed:
Add space to new setting name
